### PR TITLE
Advanced Health Scanner Fix + TGUI

### PR DESCRIPTION
**Pull request by DTraitor**
### Purpose
- Body Scanner now uses TGUI. 
- Analyse Health verb now uses same TGUI interface as body scanner.
- Added advanced health scanner as high end tech in bio tech tree. It can also be bought in store.
- Printed in body scanner health report displays info again.
image

### CODE REVIEW

**Comment by DTraitor** [comment]
Damn, it seems Map Diff check error isn't connected to this PR
```
Octokit.ApiException: An error occurred with this API request
   at Octokit.Connection.HandleErrors(IResponse response) in /home/runner/work/octokit.net/octokit.net/Octokit/Http/Connection.cs:line 691
   at Octokit.Connection.RunRequest(IRequest request, CancellationToken cancellationToken) in /home/runner/work/octokit.net/octokit.net/Octokit/Http/Connection.cs:line 668
   at Octokit.Connection.Run[T](IRequest request, CancellationToken cancellationToken) in /home/runner/work/octokit.net/octokit.net/Octokit/Http/Connection.cs:line 652
   at Octokit.ApiConnection.Patch[T](Uri uri, Object data, String accepts)
   at MapDiffBot.Core.GitHubManager.UpdateCheckRun(Int64 repositoryId, Int64 checkRunId, CheckRunUpdate checkRunUpdate, CancellationToken cancellationToken)
   at MapDiffBot.Core.PayloadProcessor.<>c__DisplayClass14_2.<<GenerateDiffs>g__RunNext|6>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at MapDiffBot.Core.PayloadProcessor.<>c__DisplayClass14_2.<<GenerateDiffs>g__RunNext|6>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at MapDiffBot.Core.PayloadProcessor.<>c__DisplayClass14_2.<<GenerateDiffs>g__RunNext|6>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at MapDiffBot.Core.PayloadProcessor.<>c__DisplayClass14_2.<<GenerateDiffs>g__RunNext|6>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at MapDiffBot.Core.PayloadProcessor.<>c__DisplayClass14_2.<<GenerateDiffs>g__RunNext|6>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at MapDiffBot.Core.PayloadProcessor.<>c__DisplayClass14_2.<<GenerateDiffs>g__RunNext|6>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at MapDiffBot.Core.PayloadProcessor.<>c__DisplayClass14_2.<<GenerateDiffs>g__RunNext|6>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at MapDiffBot.Core.PayloadProcessor.<>c__DisplayClass14_2.<<GenerateDiffs>g__RunNext|6>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at MapDiffBot.Core.PayloadProcessor.<>c__DisplayClass14_2.<<GenerateDiffs>g__RunNext|6>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at MapDiffBot.Core.PayloadProcessor.<>c__DisplayClass14_2.<<GenerateDiffs>g__RunNext|6>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at MapDiffBot.Core.PayloadProcessor.<>c__DisplayClass14_2.<<GenerateDiffs>g__RunNext|6>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at MapDiffBot.Core.PayloadProcessor.GenerateDiffs(PullRequest pullRequest, Int64 checkRunId, IReadOnlyList`1 changedDmms, IServiceScope scope, CancellationToken cancellationToken)
   at MapDiffBot.Core.PayloadProcessor.ScanPullRequestImpl(Int64 repositoryId, Int32 pullRequestNumber, IServiceScope scope, CancellationToken cancellationToken)
```

File: code/game/sound.dm [hunk 1]
```diff
@@ -65,7 +65,7 @@ GLOBAL_LIST_INIT(bubble_small_sound,list('sound/machines/tanksmallbubble1.ogg','
 GLOBAL_LIST_INIT(fleshtear_sound, list('sound/effects/organic/flesh_tear_1.ogg','sound/effects/organic/flesh_tear_2.ogg','sound/effects/organic/flesh_tear_3.ogg',))
 
 
-/proc/playsound(var/atom/source, soundin, vol as num, vary, extrarange as num, falloff, var/is_global, var/frequency, var/is_ambiance = 0)
+/datum/proc/playsound(var/atom/source, soundin, vol as num, vary, extrarange as num, falloff, var/is_global, var/frequency, var/is_ambiance = 0)
```
Comment (NanakoAC) [inline code/game/sound.dm]: this is a huge change which seems questionable
Datums don't exist in physical space, so they can't emit sounds in a meaningful way. They could be responsible for emitting sounds, but i don't believe there's any need to track that outside of looped audio

I'd be surprised if this doesn't break something. If there's a need for it please explain

File: code/modules/clothing/spacesuits/rig/modules/ds13/healthbar.dm [hunk 2]
```diff
@@ -36,7 +36,8 @@
 		tracking_level = pick(RIG_SENSOR_OFF, RIG_SENSOR_BINARY, RIG_SENSOR_VITAL, RIG_SENSOR_TRACKING)
 
 /obj/item/rig_module/healthbar/Destroy()
-	unregister_user()
```
Comment (NanakoAC) [inline code/modules/clothing/spacesuits/rig/modules/ds13/healthbar.dm]: doesnt that proc already do safety checks?

**Review by NanakoAC — CHANGES_REQUESTED** [review]
two concerns

File: code/modules/clothing/spacesuits/rig/modules/ds13/healthbar.dm (hunk 2 above)
Comment (DTraitor) [reply code/modules/clothing/spacesuits/rig/modules/ds13/healthbar.dm]: Nope, but I think it'll be better to move this check to unregister_user()

File: code/game/machinery/adv_med.dm [hunk 3]
```diff
@@ -153,258 +153,453 @@
...
-			dat += "<br><HR><span class='warning'>This device can only scan compatible lifeforms.</span>"
+		return TRUE
+
+	. = TRUE
+	switch(action)
+		if("ejectify")
+			BS.eject()
+		if("print_p")
+			BS.visible_message("<span class='notice'>[BS] rattles and prints out a sheet of paper.</span>")
+			BS.playsound(src, 'sound/machines/printer.ogg', 50, 1)
```
Comment (NanakoAC) [inline code/game/machinery/adv_med.dm]: change this to 
playsound(BS, 'sound/machines/printer.ogg', 50, 1)

**Review by NanakoAC — APPROVED** [review]

**Pull request merged**